app/lib/package/backend.dart

```diff
@@ -1014,10 +1022,19 @@ class PackageBackend {
     required String name,
     required AuthenticatedAgent agent,
   }) async {
-    final isGoogleComUser =
-        agent is AuthenticatedUser && agent.user.email!.endsWith('@google.com');
-    final isReservedName = matchesReservedPackageName(name);
-    final isExempted = isGoogleComUser && isReservedName;
+    final reservedPackage = await lookupReservedPackage(name);
+    final reservedEmails = reservedPackage?.emails ?? const <String>[];
+
+    bool isAllowedUser = false;
+    if (agent is AuthenticatedUser) {
+      final email = agent.user.email;
+      isAllowedUser = email != null &&
+          (email.endsWith('@google.com') || reservedEmails.contains(email));
```

Comment:
I suggest that now we can allow list specific emails, we don't do this general thing for `@google.com` emails.

It's much better to be specific, this is generally rare.